src/adagrams.js

```diff
@@ -1,8 +1,166 @@
-const Adagrams = {
-  drawLetters() {
+class Adagrams {
+  
+  static drawLetters() {
     // Implement this method for wave 1
-  },
-};
+    const letters = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1,
+    }
+
+    let chosen = {};
+    let drawn = [];
+    //ASCII for A-Z, using ASCII to draw letters to reduce time complexity for having loop inside loop
+    const min = 65;
+    const max = 90;
+    const range = max - min + 1;
+    let numOfWords = 0;
+    
+    while (numOfWords < 10) {
+      let num = min + Math.floor(Math.random() * range);
+      let letter = String.fromCharCode(num);
+      if (chosen[letter]) {
+        if (chosen[letter] < letters[letter]) {
+        chosen[letter] += 1;
+        drawn.push(letter);
+        numOfWords += 1;
+        }
+      } else {
+        chosen[letter] = 1;
+        drawn.push(letter);
+        numOfWords += 1;
+      }
+    }
+      return drawn;
+    }
+  
+  static usesAvailableLetters(input, lettersInhand) {
+    let letterCount = {};
+    lettersInhand.forEach((char) => { if (letterCount[char]){
+      letterCount[char] += 1 ;
+    } else {
+      letterCount[char] = 1;
+    } })
```

Comment:
Incorrect indentation (putting the body of the function passed to `forEach` on its own line makes this clearer.):

```suggestion
    lettersInhand.forEach((char) => { 
      if (letterCount[char]) {
        letterCount[char] += 1;
      } else {
        letterCount[char] = 1;
      } 
    })
```